avoidance/src/common.cpp

```diff
@@ -315,4 +393,105 @@ void fillUnusedTrajectoryPoint(mavros_msgs::PositionTarget& point) {
   point.yaw = NAN;
   point.yaw_rate = NAN;
 }
+
+// This function is a refactor of the original in the pcl library
+pcl::PointCloud<pcl::PointXYZ> removeNaNAndGetMaxima(pcl::PointCloud<pcl::PointXYZ>& cloud) {
```

Comment:
Depends on whether the global planner would also be interested in receiving NAN-filtered point clouds?

Reply:
yes I agree that this function could be useful as well for global planner. I think it can stay where it is for now. I just think that once we have more of these bigger functions in common.cpp it might make sense to split that file again into super basic stuff and the more specialized functions

Reply:
Yes, I thought about a FOV class with respective member functions. Not sure if it's worth it